File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/DeviceUseStatement.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class DeviceUseStatement : DomainResource, IResource
{
    public Base.ResourceReference[]? DerivedFrom { get; set; }
    public Base.Period? TimingPeriod { get; set; }
    public Base.CodeableConcept[]? ReasonCode { get; set; }
    public Base.ResourceReference? Source { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public string? TimingDateTime { get; set; }
    public string? TimingTiming { get; set; }
    public required string Status { get; set; }
    public object? Timing    
    {
        get
        {
            if (TimingPeriod is not null)
            {
                return TimingPeriod;
            }
    
            if (TimingDateTime is not null)
            {
                return TimingDateTime;
            }
    
            if (TimingTiming is not null)
            {
                return TimingTiming;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.Period))
            {
                TimingPeriod = (Base.Period)value;
                return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                TimingDateTime = (string)value;
                return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                TimingTiming = (string)value;
                return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public string? RecordedOn { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.CodeableConcept? BodySite { get; set; }
    public required Base.ResourceReference Device { get; set; }
    public Base.ResourceReference[]? BasedOn { get; set; }
    public required Base.ResourceReference Subject { get; set; }
    public Base.ResourceReference[]? ReasonReference { get; set; }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/DeviceUseStatement.py
from typing import Optional, List
from base import Annotation
from base import Period
from base import CodeableConcept
from base import Timing
from base import DomainResource
from base import Reference
from base import Identifier

class DeviceUseStatement(DomainResource):
    derived_from: Optional[List[Reference]] = None
    timing_period: Optional[Period] = None
    reason_code: Optional[List[CodeableConcept]] = None
    source: Optional[Reference] = None
    note: Optional[List[Annotation]] = None
    timing_date_time: Optional[str] = None
    timing_timing: Optional[Timing] = None
    status: str
    recorded_on: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    body_site: Optional[CodeableConcept] = None
    device: Reference
    based_on: Optional[List[Reference]] = None
    subject: Reference
    reason_reference: Optional[List[Reference]] = None